db/migrations/000004_ips.down.sql

```diff
@@ -0,0 +1 @@
+DROP TABLE ips;
```

Comment:
Out of curiosity...
Let's say we are at schema version 4 in the DB (i.e., we've already migrated to this version), and we want to perform a migration to bump up to version 5... what happens if we accidentally say input `up 3` instead of `up 5`?  That means we'd be reverting from version 4 to version 3.  In that case, would the version 4 `down` script be executed (dropping the `ips` table)?

Reply:
Yes, that's by design in the migration strategy that there's a rollback option.  You have a good point that this time, a mistake would mean loss of data.

What do you think about tweaking the Jenkins job that applies these changes to the persistent AWS environments?  We could:
- Add a job argument (REVERT?  ROLLBACK?) that would signify your intent to roll back the database version
- Add a step early in the job that queried for the current schema version (the `migrate` utility can do this)
- If the current schema version was higher than the version to migrate to, stop the job unless the ROLLBACK argument was set

Reply:
Yes, I think we should include that level of safety "just in case".

Reply:
👍 I've added BCDA-2416 for that enhancement.